File: Python/two_sum.py
class Solution:
    def two_sum(self, nums: list[int], target: int) -> list[int]:
        num_map = {}
        n = len(nums)

        for i in range(n):
            complement = target - nums[i]
            if complement in num_map:
                return [num_map[complement], i]
            num_map[nums[i]] = i

        return []
    

File: Go/twoSum.go
func twoSum(nums []int, target int) []int {
    myMap := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		if index, exists := myMap[complement]; exists {
            return []int{index, i}
		}
		myMap[num] = i
    }
    return nil
}